internal/runners/activate/activation.go

```diff
@@ -19,6 +20,20 @@ import (
 	"github.com/ActiveState/cli/pkg/projectfile"
 )
 
+type activationError struct {
+	error
+}
+
+func (ae activationError) IsFromActivatedShell() {}
+
+// Is Silent returns true if the wrapped error is an exit code error
+func (ae activationError) IsSilent() bool {
+	var eerr interface{ ExitCode() int }
+	return errors.As(ae, &eerr)
+}
```

Comment:
This seems too presumptuous. Because an error specifies an ExitCode it is therefore "silent"? This doesn't feel right. Also why do we need both `IsFromActivatedShell` and `IsSilent`?

Reply:
I think you are right that we could instead just implement another `ErrorCode()` function that returns zero.

The logic is necessary to implement the "implied" AC that we should not print error messages on non-zero exit codes in the subshell.  Other subshell errors (that are unlikely but could be caused by not finding the shell command or something like this) should IMO still produce error messages.